src/ShapeCrawler/Drawing/CroppingFrame.cs

```diff
@@ -0,0 +1,77 @@
+using System;
+using DocumentFormat.OpenXml;
+using ShapeCrawler.Exceptions;
+using A = DocumentFormat.OpenXml.Drawing;
+
+namespace ShapeCrawler.Drawing;
+
+/// <summary>
+///     The portion within an image which is visible. 
+/// </summary>
+/// <remarks>
+///     Values are a fraction of the whole image, e.g. `0.25` means that 25% of the image is removed
+///     on the given edge.
+///     An image which is fully visible will have 0 for all values.
+/// </remarks>
+/// <param name="left">Portion of image along left edge of source picture which will not be displayed.</param>
+/// <param name="right">Portion of image along right edge of source picture source picture which will not be displayed.</param>
+/// <param name="top">Portion of image from top edge of source picture source picture which will not be displayed.</param>
+/// <param name="bottom">Portion of image from bottom edge of source picture source picture which will not be displayed.</param>
+public record CroppingFrame(decimal left, decimal right, decimal top, decimal bottom)
+{
+    /// <summary>
+    ///     Set the cropping frame values onto the supplied source rectangle.
+    /// </summary>
+    /// <param name="aSrcRect">Rectange to be updated with our values.</param>
+    public void UpdateSourceRectangle(A.SourceRectangle aSrcRect)
+    {
+        aSrcRect.Left = ToHundredThousandths(this.left);
+        aSrcRect.Right = ToHundredThousandths(this.right);
+        aSrcRect.Top = ToHundredThousandths(this.top);
+        aSrcRect.Bottom = ToHundredThousandths(this.bottom);        
+    }
+
+    /// <summary>
+    ///     Parse a string value into a cropping frame.
+    /// </summary>
+    /// <param name="input">All four frame values separated by commas.</param>
+    /// <returns>Parsed frame.</returns>
+    public static CroppingFrame Parse(string input)
```

Comment:
I'm in favor of avoiding the use of public statics whenever possible, which is a bad thing if you're trying to write in OOP style. Such not complicated parsing can be left for the library's client code.

Then using the API as shown below, simple, I think:

```C#
[TestCase("059_crop-images.pptx", 1, "None", 0, 0, 0, 0)]
public void Crop_getter_gets_expected_values(
    string file, 
    int slideNumber, 
    string pictureName, 
    int expectedLeftCrop, 
    int expectedRightCrop, 
    int expectedTopCrop, 
    int expectedBottomCrop)
{
    // Arrange
    var pres = new Presentation(StreamOf(file));
    var picture = pres.Slide(slideNumber).Picture(pictureName);

    // Act
    var crop = picture.Crop;

    // Assert
    crop.Left.Should().Be(expectedLeftCrop);
    crop.Right.Should().Be(expectedRightCrop);
    crop.Top.Should().Be(expectedTopCrop);
    crop.Bottom.Should().Be(expectedBottomCrop);
}

[TestCase(0, 0, 0, 0)]
public void Crop_setter_sets_expected_values(int leftCrop, int rightCrop, int topCrop, int bottomCrop)
{
    // Arrange
    var pres = new Presentation(StreamOf("059_crop-images.pptx"));
    var picture = pres.Slide(1).Picture("None");

    // Act
    picture.Crop = new CroppingFrame(leftCrop, rightCrop, topCrop, bottomCrop);

    // Assert
    picture.Crop.Left.Should().Be(leftCrop);
    picture.Crop.Right.Should().Be(rightCrop);
    picture.Crop.Top.Should().Be(topCrop);
    picture.Crop.Bottom.Should().Be(bottomCrop);
}
```

What do you think?

Reply:
If that's how you prefer it, then that's how we shall do it.

BTW, you did take the easy route with 0,0,0,0. What would you do with 0.1,0.2,0.3,0.4? TestCase can't send in decimals, so now either we have to send in ints and divide them or strings and convert them. Now, we have to repeat that pattern in both tests and any future tests. So I thought it was better to DRY this and put this conversion in just one place. Also I'm a fan of hiding messy details.

Second, this same pattern follows with `FromSourceRectangle`. What would you prefer there? Constructing out of components means we have to push the per-cent-mille (hundredthousanths) conversions up to the caller, which seems like needless exposure of inner workings.

If it's OOP strictness you're after, we could have three different constructors instead. One fully qualified, one with a string, and one with a source rectangle.

I'd also point out that C# uses static `Type.Parse(string)` routinely, so I'd expect it's a pattern that developers are accustomed to working with.

Reply:
> I'd also point out that C# uses static Type.Parse(string) routinely, so I'd expect it's a pattern that developers are accustomed to working with.

I agree. Perhaps such long-established things should be preferred over the desire for strict compliance with OOP🤔

> If it's OOP strictness you're after, we could have three different constructors instead. One fully qualified, one with a string, and one with a source rectangle.

This sounds good, however, except constructor for the source rectangle. I think you mean `A.SourceRectangle`? We've done this in the code already for another feature and I think it's wrong and should also be removed as it violates the basic idea of creating an object model over Open XML SDK. I mean, the _public_ APIs should not deal with entities of Open XML SDK.